manifests/users.pp

```diff
@@ -0,0 +1,63 @@
+# This class takes two `Hashes` of user resources, one to remove, and one to
+# install.
+#
+# After the entire puppet catalog has been compiled, it will process both lists
+# and, for any resource that is not already defined in the catalog, it will
+# take the appropriate action.
+#
+# An exception will be raised if you list the same user in both lists.
+#
+# @param $remove
+#   A list of users to remove.
+#
+#   * A `Hash` can be used to add extra attributes for the user, but the
+#     `ensure` attribute will be overwritten if it is included.
+#
+# @param $install
+#   A list of users to install.
+#
+#   * A `Hash` can be used to add extra attributes for the user, but the
+#     `ensure` attribute will always be set to `$user_ensure`.
+#
+# @param $default_options
+#   A `Hash` of options to apply to all users (both remove and install.
+#   If ensure is entered in these options it will be overwritten.
+#
+#   * These options may be anything that a Puppet `User` resource can
+#     normally accept.
+#
+# @param $mode
+#   @see `deferred_resources::mode`
+#
+# @param $log_level
+#   @see `deferred_resources::log_level`
+#
+class deferred_resources::users (
+  Variant[Hash, Array[String[1]]]      $remove          = {},
```

Comment:
What is the use case for Hash input for $remove and $install ?  Why isn't an Array sufficient?